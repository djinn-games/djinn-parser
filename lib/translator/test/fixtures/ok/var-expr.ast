{
  "type": "Program",
  "name": {
    "type": "Identifier",
    "name": "var_expressions",
    "line": 1
  },
  "body": {
    "type": "Body",
    "sentences": [
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "VarDeclaration",
          "dataType": "int",
          "id": {
            "type": "Identifier",
            "name": "x",
            "line": 3
          }
        },
        "line": 3
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "Identifier",
          "name": "x",
          "line": 4
        },
        "line": 4
      }
    ],
    "line": 2
  },
  "line": 1
}