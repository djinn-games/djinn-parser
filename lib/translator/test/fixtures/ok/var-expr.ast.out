{
  "type": "Program",
  "body": [
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "AssignmentExpression",
        "operator": "=",
        "left": {
          "type": "MemberExpression",
          "object": {
            "type": "MemberExpression",
            "object": {
              "type": "Identifier",
              "name": "DJINN"
            },
            "property": {
              "type": "Identifier",
              "name": "__scope"
            }
          },
          "property": {
            "type": "Literal",
            "value": "_x",
            "raw": "\"_x\""
          },
          "computed": true
        },
        "right": {
          "type": "Literal",
          "value": 0,
          "raw": "0"
        }
      }
    },
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "BinaryExpression",
        "operator": "+",
        "left": {
          "type": "MemberExpression",
          "object": {
            "type": "MemberExpression",
            "object": {
              "type": "Identifier",
              "name": "DJINN"
            },
            "property": {
              "type": "Identifier",
              "name": "__scope"
            }
          },
          "property": {
            "type": "Literal",
            "value": "_x",
            "raw": "\"_x\""
          },
          "computed": true
        },
        "right": {
          "type": "Literal",
          "value": 5,
          "raw": "5"
        }
      }
    }
  ]
}
